Sources/ClientRuntime/Networking/Http/HttpResponse.swift

```diff
@@ -32,6 +36,15 @@ extension HttpResponse: CustomDebugStringConvertible {
     }
 }
 
+extension HttpResponse: WireDataProviding {
+
+    public func data() async throws -> Data {
+        let data = try await body.readData()
+        body = .data(data)
+        return data ?? Data()
+    }
+}
+
```

Comment:
The `HttpResponse` type is extended with a method to get the data out of it.  This allows `SmithyReadWrite` to get data from an HTTP response without depending on the type.